Source/monster.cpp

```diff
@@ -1489,9 +1489,7 @@ void MonsterAttackPlayer(int i, int pnum, int hit, int minDam, int maxDam)
 				dam = 64;
 			int mdam = dam * (GenerateRnd(10) + 20L) / 100;
 			monster._mhitpoints -= mdam;
-			dam -= mdam;
-			if (dam < 0)
-				dam = 0;
+			dam = std::max(dam - mdam, 0);

```

Comment:
You can also clean up the `dam += player._pIGetHit` call in the same way a few lines up :)

Reply:
Updated the pull request. Thank you @ephphatha !

Reply:
There's another case of the same thing on lines 1525-1527 (a few lines above the next change)

Reply:
updated the pull request clamp another case. Thank you!